src/components/views/rooms/wysiwyg_composer/message.ts

```diff
@@ -0,0 +1,190 @@
+        });
+    }*/
+
+    return content;
+}
+
+export function sendMessage(
+    message: string,
+    { roomContext, mxClient, ...params }: SendMessageParams,
+) {
+    const { relation, replyToEvent } = params;
+    const { room } = roomContext;
+    const { roomId } = room;
+
+    const posthogEvent: ComposerEvent = {
+        eventName: "Composer",
+        isEditing: false,
+        isReply: Boolean(replyToEvent),
+        inThread: relation?.rel_type === THREAD_RELATION_TYPE.name,
+    };
+
+    // TODO thread
+    /*if (posthogEvent.inThread) {
+        const threadRoot = room.findEventById(relation?.event_id);
+        posthogEvent.startsThread = threadRoot?.getThread()?.events.length === 1;
+    }*/
+    PosthogAnalytics.instance.trackEvent<ComposerEvent>(posthogEvent);
+
+    let content: IContent;
+
+    // TODO slash comment
+
+    // TODO replace emotion end of message ?
+
+    // TODO quick reaction
+
+    if (!content) {
+        content = createMessageContent(
+            message,
+            params,
+        );
+    }
+
+    // don't bother sending an empty message
+    if (!content.body.trim()) {
+        return;
+    }
+
+    if (SettingsStore.getValue("Performance.addSendMessageTimingMetadata")) {
+        decorateStartSendingTime(content);
+    }
+
+    const threadId = relation?.rel_type === THREAD_RELATION_TYPE.name
+        ? relation.event_id
+        : null;
+
+    const prom = doMaybeLocalRoomAction(
+        roomId,
+        (actualRoomId: string) => mxClient.sendMessage(actualRoomId, threadId, content),
+        mxClient,
+    );
+
+    // TODO reply
+    /*if (replyToEvent) {
+        // Clear reply_to_event as we put the message into the queue
+        // if the send fails, retry will handle resending.
+        dis.dispatch({
+            action: 'reply_to_event',
+            event: null,
+            context: roomContext.timelineRenderingType,
+        });
+    }*/
+    dis.dispatch({ action: "message_sent" });
+    CHAT_EFFECTS.forEach((effect) => {
+        if (containsEmoji(content, effect.emojis)) {
+            // For initial threads launch, chat effects are disabled
+            // see #19731
+            const isNotThread = relation?.rel_type !== THREAD_RELATION_TYPE.name;
+            if (!SettingsStore.getValue("feature_thread") || isNotThread) {
+                dis.dispatch({ action: `effects.${effect.command}` });
+            }
+        }
+    });
+    if (SettingsStore.getValue("Performance.addSendMessageTimingMetadata")) {
+        prom.then(resp => {
+            sendRoundTripMetric(mxClient, roomId, resp.event_id);
+        });
+    }
+
+    // TODO save history
+    // TODO save local state
+
+    //if (shouldSend && SettingsStore.getValue("scrollToBottomOnMessageSent")) {
+    if (SettingsStore.getValue("scrollToBottomOnMessageSent")) {
+        dis.dispatch({
+            action: "scroll_to_bottom",
+            timelineRenderingType: roomContext.timelineRenderingType,
+        });
+    }
```

Comment:
I'm not convinced this should live in here. This looks like a side effect that should be done when catching the `message_sent` action.

Also do not think this should block this pull request from landing.

Reply:
To be honest, I copied the behaviour of the `SendMessageComposer` here. My goal here is to integrate the base of the `matrix-wysiwyg` and not change the current `sendMessage` behaviour.